linode/instancetypes/framework_models.go

```diff
@@ -0,0 +1,40 @@
+package instancetypes
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/linode/helper/frameworkfilter"
+	"github.com/linode/terraform-provider-linode/linode/instancetype"
+)
+
+type InstanceTypeFilterModel struct {
+	ID      types.String                     `tfsdk:"id"`
+	Order   types.String                     `tfsdk:"order"`
+	OrderBy types.String                     `tfsdk:"order_by"`
+	Filters frameworkfilter.FiltersModelType `tfsdk:"filter"`
+	Types   []instancetype.DataSourceModel   `tfsdk:"types"`
+}
+
+func (model *InstanceTypeFilterModel) parseInstanceTypes(ctx context.Context,
+	instanceTypes []linodego.LinodeType,
+) diag.Diagnostics {
+	result := make([]instancetype.DataSourceModel, len(instanceTypes))
+
+	for i, instanceType := range instanceTypes {
+		var m instancetype.DataSourceModel
+
+		diags := m.ParseLinodeType(ctx, &instanceType)
```

Comment:
Here we need to use `&instanceTypes[i]` to address the pointer issue `Implicit memory aliasing in for loop.`, because `instasnceType` is reused for iteration and it points to the same memory location. We can use `instanceTypes[i]` to get the actual element's address.

Reply:
Ah that makes sense. Thanks!